src/coreclr/vm/jithelpers.cpp

```diff
@@ -3642,3 +3466,250 @@ void InitJITHelpers2()
     SetJitHelperFunction(CORINFO_HELP_INIT_PINVOKE_FRAME, (void *)GenerateInitPInvokeFrameHelper()->GetEntryPoint());
 #endif // TARGET_X86 || TARGET_ARM
 }
+
+
+#if defined(TARGET_32BIT) && !(defined(TARGET_X86) && defined(TARGET_WINDOWS))
+
+// Using the musttail attribute requires that we disble our protection against using return in the wrong place
+#ifdef return
+#undef return
+#endif
```

Comment:
Related: I'm trying to remove the cases outside of HMFs that engage this infra as it continues to cause us problems in places.